File: in_Java/1232 Check If It Is a Straight Line.java
class Solution {
    public boolean checkStraightLine(int[][] coordinates) {
        if(coordinates.length == 2) return true;
        for(int i = 2; i < coordinates.length; i++){
            int x1 = coordinates[i-2][0];
            int y1 = coordinates[i-2][1];
            int x2 = coordinates[i-1][0];
            int y2 = coordinates[i-1][1];
            int x3 = coordinates[i][0];
            int y3 = coordinates[i][1];
            if((y3-y1)*(x2-x1) != (y2-y1)*(x3-x1)) return false;
        }
        return true;
    }
}

File: in_Python/1232 Check If It Is a Straight Line.py
class Solution(object):
    def checkStraightLine(self, coordinates):
        """
        :type coordinates: List[List[int]]
        :rtype: bool
        """
        a1 = coordinates[0][0]
        a2 = coordinates[0][1]
        b1 = coordinates[1][0]
        b2 = coordinates[1][1]
        for c in range(2, len(coordinates)):
            c2 = coordinates[c][1]
            c1 = coordinates[c][0]
            if (b2-a2) * (c1-a1) != (b1-a1) * (c2-a2):
                return False
        return True